common/models/templates/src/trie-model.ts

```diff
@@ -94,13 +94,21 @@ class Traversal implements LexiconTraversal {
    */
   root: Node;
 
-  constructor(root: Node, prefix: string) {
+  /**
+   * The max weight for the Trie being 'traversed'.  Needed for probability
+   * calculations.
+   */
+  totalWeight: number;
+
+  constructor(root: Node, prefix: string, maxWeight: number) {
     this.root = root;
     this.prefix = prefix;
+    this.totalWeight = maxWeight;
   }
 
   *children(): Generator<{char: string, traversal: () => LexiconTraversal}> {
     let root = this.root;
+    const totalWeight = this.totalWeight;
```

Comment:
Is this just to avoid referencing `this.totalWeight` or does `this.totalWeight` change in the loop? Can you add a comment accordingly?

Reply:
Minification can't minify `this.totalWeight` if we don't mutate the field, which we don't.  A `const totalWeight`, though... that's much easier to rename and shorten - especially given the `const`, which likely can't be inferred from `this.totalWeight`.